User_Service/Controllers/controller.go

```diff
@@ -0,0 +1,270 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	data "github.com/E-Furqan/Food-Delivery-System/Data"
+	config "github.com/E-Furqan/Food-Delivery-System/database_config"
+	"github.com/E-Furqan/Food-Delivery-System/utils"
+	"github.com/gin-gonic/gin"
+)
+
+func Register(c *gin.Context) {
+
+	var reg_data data.User
+
+	if err := c.ShouldBindJSON(&reg_data); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	if err := config.DB.Where("username = ?", reg_data.Username).First(&reg_data).Error; err == nil {
```

Comment:
separate layer for DB

Reply:
have created a separate package for DB which use dependence injection  and have all the required function related to db